syntax = "proto3";
option cc_enable_arenas = true;
package NetProto;

import "Common.proto";
import "BattleEnum.proto";

message RspFreeHero
{
	uint64 red_hero_id = 1;
	uint64 blue_hero_id = 2;
}

message SelectHeroReq
{
	uint64 hero_id = 1;
}

message SelectHeroRsp
{
	uint64 hero_id = 1;
	bool is_succ = 2;
}

message SceneObjectDisappear
{
	repeated uint64 objids = 1;
}

message MoveToPos
{
	PBVector2 pos = 1;
}

message BattleOperation
{
	EPlayerOpera opera = 1;
	uint64 target_id = 2;
	float dir = 3;
	PBVector2 pos = 4;
	ESkillSlot skill_slot = 5;
}

message ViewGrid
{
	PBVector2 center = 1;
	int32 grid_type = 2;
}

message ViewSnapshotDiff
{
	repeated int32 more_grids = 1;
	repeated int32 miss_grids = 2;
}

message ViewSnapshot
{
	repeated int32 light_grids = 1;
}

message ViewAllGrids
{
	float grid_size = 1;
	int32 row = 2;
	int32 col = 3;
	repeated ViewGrid grids = 4;
}

message SceneUnitState
{
	uint64 su_id = 1;
	ESceneUnitType unit_type = 2;
	int32 model_id = 3;
	PBVector3 pos = 4;
	float face_dir = 5;
	uint64 parent_su_id = 6;
}

message SceneUnitTransform
{
	uint64 su_id = 1;
	PBVector3 pos = 2;
	float face_dir = 3;
	uint64 parent_su_id = 4;
	int64 ms = 5;
}

message SceneUnitMove
{
	uint64 su_id = 1;
	EMoveAgentState move_agent_state = 2;

	float max_move_speed = 3;
	PBVector2 move_desired_pos = 4;
	float move_desired_dir = 5;
	int64 immobilized_end_timestamp_ms = 6;

	PBVector2 force_pos_destination = 7;
	float force_pos_speed = 8;
	
	PBVector2 force_line_velocity = 9;
	float force_line_elasped_sec = 10;
	float force_line_total_sec = 11;
	PBVector2 force_line_end_pos = 12;

	PBVector3 unit_pos = 13;
	int64 now_ms = 14;
}

message SceneUnitSkillAction
{
	uint64 su_id = 1;
	ESkillState stage = 2;
	int32 skill_id = 3;
	int64 stage_begin_ms = 4;
}

message SceneTimeSync
{
	int64 ms = 1;
	float sec = 2;
}

message BulletState
{
	uint64 su_id = 1;
	float speed = 2;
	PBVector3 target_pos = 5;
	uint64 target_suid = 6;
	int32 curr_state = 7;
	PBVector3 unit_pos = 10;
	int64 now_ms = 11;
}

message BulletTargetPos
{
	uint64 su_id = 1;
	PBVector3 target_pos = 2;
	int64 now_ms = 3;
	PBVector3 unit_pos = 4;
}

